### PR TITLE
Bug 1181290 - [Aries][Gallery] Images draw in segments from right to left when viewed fullscreen r=djf

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e61c76c4c45d658a62c63ad901d28b72b4679328

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2ee4e0c1147d799b664bc35a385eabda473e1870

File: apps/gallery/js/MetadataParser.js [hunk 1]
```diff
@@ -303,7 +303,16 @@ var metadataParser = (function() {
         // thumbnail and an external preview image. Note that progressive
         // jpegs will have a type 'pjpeg' and will be handled below
         // like png images.
-        if (metadata.type === 'jpeg') {
+
+        // For jpeg images with sizes more than 2.5 mb, we need
+        // to create preview on sdcard to avoid flash while opening image.
+        // See bug 1181290.
+        // This constraint can be removed when gecko supports downscale during
```
Comment (davidflanagan) [inline apps/gallery/js/MetadataParser.js]: We don't know for sure that we'll be able to remove the constraint. We hope so, though.

File: apps/gallery/js/MetadataParser.js [hunk 2]
```diff
@@ -303,7 +303,16 @@ var metadataParser = (function() {
...
+        // decode by canvas size for offscreen images. This will
+        // facilitate removing #moz-sample-size and using ddd.
+        // See bug 1206206.
+
+        if (metadata.type === 'jpeg' && file.size < 2.5 * 1024 * 1024) {
```
Comment (davidflanagan) [inline apps/gallery/js/MetadataParser.js]: Please use the image size, not the file size here. I'd suggest 5 or 8 megapixels as the limit, not 2.5 megabytes:
```
if (metadata.type === 'jpeg' && metadata.width * metadata.height < 5 * 1024 * 1024) {
```

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=72affe6b076ea62b277b0552dd4cf263709cbce4

**Pull request merged**